sklearn/model_selection/_split.py

```diff
@@ -1139,9 +1132,12 @@ def split(self, X, y=None, groups=None):
             The testing set indices for that split.
         """
         n_repeats = self.n_repeats
-        rng = check_random_state(self.random_state)
+        rng = check_random_state(self._seed)
 
-        for idx in range(n_repeats):
+        for _ in range(n_repeats):
+            # the local rng object will be consumed exactly once in
+            # self.cv.__init__, hence ensuring that each repetition yields
+            # different folds.
```

Comment:
That's a clear change of semantics of what "random_state" means. I am not very comfortable with it, because it means that semantics of arguments vary across the library, which is bad for usability.